opal/core/pathway/steps.py

```diff
@@ -16,9 +16,9 @@ def delete_others(data, model, patient=None, episode=None):
     from opal.models import EpisodeSubrecord, PatientSubrecord
 
     if issubclass(model, EpisodeSubrecord):
-        existing = model.objects.filter(episode=episode)
+        existing = model.objects.for_episode(episode=episode)
```

Comment:
if we did filter rather than for_episode here then we would delete all of the things even if we had puposefully excluded them using the `for_episode` method